kind/macos-kind.sh

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env bash
```

Comment:
Why did we have to make a completely separate copy for mac?    Was there no way to conditionalize the mac parts inside the script?    It'll be a hassle maintaining two separate scripts and keeping them functionally in sync... would be better if there was just one.

Reply:
Due to the high number of modifications in the script, We decided to make a separate copy for mac. There is a way to conditionalize the mac parts inside the script.

Reply:
First, it doesn't look like an especially high number of mods.    I see 14 total changes.    Running macos-kind.sh on Ubuntu generally works (except for the check for colima... which I don't see used by the script isn't obviously used to me?).   For each change:

- getopts change works on linux
- if check after getops is ok on linux
- assuming the colima check is needed, set the OS to "mac" and make it conditional
- The various "sleeps" should ideally be replaced by "wait_for_..." but if you want to leave them just pick a number that is good for both
- Looks like the sed issue is just about:
       sed -i '' ...
   ...vs...
       sed ...
   So why not do something like:
     (at the top) BSDSED=" "
     (inside the macos plaform check make it two signle quotes inside the double quotes) BSDSED="''"
   Then make each sed line be:
       sed -i $BSDSED ...

Not sure what the "14i" sed for 14i is for?   Can we fix that to work for both?